### PR TITLE
CN 313 Update B&R agent

### PR DESCRIPTION
**Pull request by cagric0**
In addition to S3, GS and Azure are also supported for the backup and restore.

### CODE REVIEW

File: config/samples/backup.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+apiVersion: hazelcast.com/v1alpha1
```
Comment (hasancelik) [inline config/samples/backup.yaml:1]: Could you rename this file as `_v1alpha1_hotbackup_agent.yaml` then remove exsiting one?

File: controllers/hazelcast/hot_backup_controller.go [hunk 2]
```diff
@@ -135,6 +135,10 @@ func (r *HotBackupReconciler) Reconcile(ctx context.Context, req reconcile.Reque
 	}
 
 	if hb.Spec.BucketURL != "" {
+		if hb.Spec.SecretName == "" {
```
Comment (hasancelik) [inline controllers/hazelcast/hot_backup_controller.go:138]: This logic should be moved to [validation.go](https://github.com/hazelcast/hazelcast-platform-operator/blob/main/controllers/hazelcast/validation/validation.go)

File: config/samples/backup.yaml (hunk 1 above)
Comment (cagric0) [reply config/samples/backup.yaml:1]: Is it okay to keep commented buckets and secrets?

**Review by dzeromski-hazelcast — APPROVED** [review]

File: config/samples/backup.yaml (hunk 1 above)
Comment (hasancelik) [reply config/samples/backup.yaml:1]: I think so 👍

**Pull request merged**